pkg/client/src/app/pages/applications/analysis-wizard/custom-rules.tsx

```diff
@@ -97,7 +96,7 @@ export const CustomRules: React.FunctionComponent<ICustomRules> = ({
     });
 
     return rules.flat();
-  }, [customRulesFiles]);
+  }, [customRulesFiles, sources, setValue]);
```

Comment:
Why add the set value handler here?

Reply:
Because it's used above in the useMemo().

Reply:
isn't it already defined in the enclosing scope? No need to add the function definition to this array as it is not ever changing.

Reply:
I get an error if we don't:
`React Hook React.useMemo has a missing dependency: 'setValue'. Either include it or remove the dependency array.eslintreact-hooks/exhaustive-deps`